x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -285,12 +285,17 @@ export const createMetricThresholdExecutor = (libs: InfraBackendLibs) =>
             ? WARNING_ACTIONS_ID
             : FIRED_ACTIONS_ID;
 
-        const additionalContext = hasAdditionalContext(params.groupBy, validGroupByForContext)
+        let additionalContext = hasAdditionalContext(params.groupBy, validGroupByForContext)
           ? alertResults && alertResults.length > 0
             ? alertResults[0][group].context
             : null
           : null;
 
+        if (!additionalContext) additionalContext = { tags: [] };
+        additionalContext.tags = additionalContext.tags
+          ? [...additionalContext.tags, ...options.rule.tags]
```

Comment:
I don't think you want to concatenate the lists, in case there are duplicates.  Unless they are removed somewhere else?  What you could do is `additionalContext.tags = Array.from(new Set(additionalContext.tags))`, as an additional line, after this statement.